README.md

```diff
@@ -34,12 +34,117 @@ changes to reuse logic doesn't break any of the migrations using the common logi
 
 # How to create an upgrade
 
-* `cp -r template <okctl target version>`, where `<okctl target version>` is explained under [Binaries](#binaries).
+This section describes a workflow for developing, testing and releasing an upgrade. 
+
+## Create the upgrade
+
+* `cp -r template <okctl target version>`, where `<okctl target version>` is explained under [Upgrade binaries](#upgrade-binaries).
     * Example: `cp -r template 0.0.60.some-component`
 * Edit the upgrade to your needs (:information_source: Tip: start with `upgrade.go`)
-* Optional: To have a look at the release files before publishing, run `make release-test UPGRADE_VERSION=0.0.65.some-component`,
-  which will create a `dist` directory.
-* To make the actual release, first push the upgrade to the main branch (through a PR, preferrably). Then run the following:
+
+## Test the upgrade
+
+### Test continuously while developing
+
+You need two things for running the upgrade directly (i.e. not using `okctl upgrade`).
+
+* Environment variables
+
+Every upgrade must run with the same environment as `okctl show credentials` (which equals `okctl venv`).
+
+So if running through an IDE such as IntelliJ, add environmental variables to be at least those from `okctl show credentials`.
```

Comment:
This could probably benefit from `okctl show credentials > .env.local` and a description on how to use the env file in intellij

Reply:
Might have to ensure only stdout or stderr for that one. Can't remember which one printed only the commands